sklearn/utils/fixes.py

```diff
@@ -393,3 +393,47 @@ def array_equal(a1, a2):
         return bool(np.asarray(a1 == a2).all())
 else:
     from numpy import array_equal
+
+if np_version < (1, 10):
+
+    def norm(X, ord=None, axis=None, keepdims=False):
+        """
+        Handles the axis and keepdims parameters for the norm function in
+        old versions of numpy.
+        WARNING: Only tested for 2 dimensions, do not use with 1 dimensional arrays
+        """
+
+        if X.ndim == 1:
+            return _norm_1d(X, ord=ord, keepdims=keepdims)
+
+        if axis is None:
+            norm = np.linalg.norm(X, ord=ord)
+            if keepdims:
+                norm = norm.reshape(1, 1)
+            return norm
+
+        if axis == 0:
+            X = X.T
+
+        norm = np.zeros(X.shape[0])
+        for i in range(len(norm)):
```

Comment:
Yes but why and where ? Is the way I chose to implement it inneficient ? I chose to use `np.linalg.norm` on each line so that I don't have to bother with the `ord` parameter, I don't see how I could use enumerate without having to  write one case for each possible `ord`.